bus-mapping/src/circuit_input_builder/transaction.rs

```diff
@@ -183,6 +183,8 @@ pub struct Transaction {
     pub id: u64,
     /// The raw transaction fields
     tx: geth_types::Transaction,
+    /// Invalid tx
+    pub invalid: bool,
```

Comment:
A optimisation suggestion: maybe do not need to define new field `invalid`, because at the place we need to derived logic based on `tx.invalid`  can just use `geth_trace.invalid` instead ?